nitro_rpc/src/NitroRPC.sol

```diff
@@ -71,12 +72,6 @@ contract NitroRPC is IForceMoveApp {
             )
         );
         
-        // Apply the Ethereum Signed Message prefix.
-        bytes32 ethSignedMessageHash = keccak256(
-            abi.encodePacked("\x19Ethereum Signed Message:\n32", messageHash)
-        );
-        
-        // Recover the signer address using ecrecover.
-        return ecrecover(ethSignedMessageHash, signature.v, signature.r, signature.s);
+        return ECDSA.recover(MessageHashUtils.toEthSignedMessageHash(messageHash), signature.v, signature.r, signature.s);
```

Comment:
There should be an ECDSA version of `recover` that takes `signature` in form of `bytes` and do v,r,s separation on its own

Reply:
Also, you can calculate `messageHash` as `keccak256(abi.encode(payload))`.